src/pq_flash_index.cpp

```diff
@@ -425,13 +432,130 @@ namespace diskann {
     }
   }
 
+  template<typename T, typename LabelT>
+  inline int32_t PQFlashIndex<T,LabelT>::get_filter_number(
```

Comment:
It looks like that, but looking at the code someone intentionally made it return int32_t. Also, we can use `std::find_if(_filter_list.begin(), _filter_list.end(), []const LabelT label) {return label == filter_label});` instead of using raw loops.

Reply:
Are the labels here raw values or their converted int values?

Reply:
This function is where the raw label values are being converted into their internally stored values

Reply:
Actually, I believe LabelT refers to the raw type, `uint` or `ushort`. This should be updated